services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/file/FileHandleManagerImpl.java

```diff
@@ -409,36 +416,38 @@ private String getS3SignedUrlForS3FileHandle(S3FileHandle handle) {
 
 		ResponseHeaderOverrides responseHeaderOverrides = new ResponseHeaderOverrides();
 
-		String contentType = handle.getContentType();
-		if (StringUtils.isNotEmpty(contentType) && !NOT_SET.equals(contentType)) {
-			responseHeaderOverrides.setContentType(contentType);
-		}
-		String fileName = handle.getFileName();
-		if (StringUtils.isNotEmpty(fileName) && !NOT_SET.equals(fileName)) {
-			responseHeaderOverrides.setContentDisposition(ContentDispositionUtils.getContentDispositionValue(fileName));
-		}
+		Optional<String> contentType = getFileHandleResponseContent(handle).get(RESPONSE_CONTENT_TYPE);
+		contentType.ifPresent(type -> responseHeaderOverrides.setContentType(type));
+
+		Optional<String> fileName = getFileHandleResponseContent(handle).get(RESPONSE_CONTENT_DISPOSITION);
+		fileName.ifPresent(name -> responseHeaderOverrides.setContentDisposition(ContentDispositionUtils.getContentDispositionValue(name)));
 
 		request.setResponseHeaders(responseHeaderOverrides);
 		return s3Client.generatePresignedUrl(request).toExternalForm();
 	}
 
 	private String getCloudFrontSignedUrlForS3FileHandle(S3FileHandle handle) {
-		String cloudFrontPrivateKey = config.getCloudFrontPrivateKey();
-		String keyPairId = config.getCloudFrontKeyPairId();
-		String distributionDomainName = config.getCloudFrontDomainName();
-		PrivateKey privateKey = KeyPairUtil.getPrivateKeyFromPEM(cloudFrontPrivateKey, RSA);
+		String keyPairId = cloudFrontCache.getKeyPairId();
+		String distributionDomainName = cloudFrontCache.getDomainName();
+		PrivateKey privateKey = cloudFrontCache.getPrivateKey();
 		Date expirationDate = new Date(System.currentTimeMillis() + PRESIGNED_URL_EXPIRE_TIME_MS);
 
 		String resourceUrl;
+
+		URIBuilder resourceUrlBuilder = new URIBuilder()
+				.setScheme("https")
+				.setHost(distributionDomainName)
+				.setPath(handle.getKey());
+
+		Optional<String> contentType = getFileHandleResponseContent(handle).get(RESPONSE_CONTENT_TYPE);
```

Comment:
we can iterate over the map returned by getFileHandleResponseContent and add values for all the parameters